app/controllers/application_controller.rb

```diff
@@ -178,6 +178,8 @@ def update_cart # rubocop:disable MethodLength, AbcSize
     # reload appropriate divs / exit
     prepare_catalog_index_vars if params[:controller] == 'catalog'
 
+    # skip renders when called by submit_cart_updates_form
+    return if params[:form]
```

Comment:
Hmmm, looking at this, I think you'd refactor so that the general cart update code was in one method and then you'd have a separate method for each of the controller actions - one for the old `update_cart`, and one for `submit_cart_updates_form`. This way, the functional / backend changes are all handled by a single method, and then the controller-specific code is separated into two different methods. Does that make sense? This way is pretty brittle and specific to the way you structured the form HTML - we don't want `update_cart` to know anything about where it's being called from, I don't think. Let me know if you have any questions!